build/LocalizationTasks/SaveXlfToResx.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using xlflib;
+
+namespace Microsoft.Build.LocalizationTasks
+{
+    public class SaveXlfToResx : Task
+    {
+        [Required]
+        public string ResxPath { get; set; }
+
+        [Required]
+        public string XlfPath { get; set; }
+
+        /// <summary>
+        /// When set to "true" it makes the task replace the resource values with another value.
+        /// Useful for testing localized builds.
+        /// 
+        /// If <see cref="TestString"/> is non empty, resource values get replaced with it.
+        /// If it is empty, resource values get replaced with !resource_id!english_resource!localized_resource!
+        /// </summary>
+        public string ReplaceWithTestString { get; set; }
```

Comment:
Kind of, because I have to edit the in-memory xlf and then ask it to serialize to resx. If I take it out, it will actually be the entire thing and this task would remain a copy but without the testing code in it. 

The alternative is to implement the localization testing functionality in a separate task and have it parse the generated resx and edit that. But that's more work. Do you think the testing functionality should be done at the resx level, via resx resource readers / writers?